sql/analyzer/indexed_joins.go

```diff
@@ -886,3 +904,27 @@ func getTablesOrSubqueryAliases(node sql.Node) []NameableNode {
 
 	return tables
 }
+
+// hasIndexableChild validates whether the join tree
+// has indexable tables.
+func hasIndexableChild(node plan.JoinNode) bool {
+	switch n := node.Right().(type) {
+	case *plan.ResolvedTable, *plan.TableAlias:
+		return true
+	case *plan.CrossJoin, *plan.ValueDerivedTable, *plan.SubqueryAlias, *plan.StripRowNode:
```

Comment:
If you're using empty case labels for control flow like this, note it with a comment